packages/flutter/lib/src/rendering/image.dart

```diff
@@ -85,8 +85,11 @@ class RenderImage extends RenderBox {
   ui.Image? get image => _image;
   ui.Image? _image;
   set image(ui.Image? value) {
-    if (value == _image)
+    if (value == _image || value != null && _image != null && value.isCloneOf(_image!)) {
```

Comment:
Why dispose when `value == _image`? Wouldn't this potentially dispose the object we are holding onto in `_image`?

Reply:
Ah, you are right. I'll add a test for this and fix it.